# Benchmarks

In [14]:
import matplotlib.pyplot as plt
import sys, os, re, commands
import numpy as np
mN = 1634.0
hbarc = 197.3161329
print('h2m   = ',hbarc**2/mN)
print('2/h2m = ',2*mN/hbarc**2)

('h2m   = ', 23.827207039541292)
('2/h2m = ', 0.08393765986424663)


### $$\sum\limits_{i<j}^\texttt{N}
\left(C^\text{Wigner}_\Lambda+C^\text{Bartlett}_\Lambda\hat{\mathcal{O}}_\text{Bartlett}\right)\cdot
e^{-\frac{\Lambda^2}{4}\left[\vec{r}(i)-\vec{r}(j)\right]^2}+
D_\Lambda\cdot\sum\limits_\text{cyc.}\sum\limits_{i<j<k}
e^{-\frac{\Lambda^2}{4}\left[\vec{r}(i)-\vec{r}(j)\right]^2}
e^{-\frac{\Lambda^2}{4}\left[\vec{r}(i)-\vec{r}(k)\right]^2}$$

$\texttt{h2m}=\frac{\hbar^2c^2}{m_N}=\frac{197.3161329\,\text{MeV}^2\cdot\text{fm}^2}{1634\,\text{MeV}}=23.83\,\text{MeV}\cdot\text{fm}^2$

$\Lambda=4\,\text{fm}^{-1}$

$C^\text{Wigner}_\Lambda=-396.568\,$MeV

$C^\text{Bartlett}_\Lambda=-8.061\,$MeV

$D_\Lambda=-8.061\,$MeV

In [8]:
status, output = commands.getstatusoutput('./svm.x np_triplet_4_806')
outp = float([ line for line in open('./output/np_triplet_4_806.txt')][-2].split('E=')[1])
print('E(0,S=1) = %4.4f MeV'%outp)
status, output = commands.getstatusoutput('./svm.x np_singlet_4_806')
outp = float([ line for line in open('./output/np_singlet_4_806.txt')][-2].split('E=')[1])
print('E(0,S=0) = %4.4f MeV'%outp)

E(0,S=1) = -19.3656 MeV
E(0,S=0) = -15.7653 MeV


### $$\nu\sum_i^N\left(x^2(i)+y^2(i)+z^2(i)\right)$$

$\frac{\vec{p}^2}{2m}+\frac{m\omega}{2}\vec{r}^2\;\to\;E_n=\hbar\omega\left(n+\frac{3}{2}\right)$
and hence we must obtain for the *__ground state__* of $N$ mass $m_N$ particles:
$$E_0:=\sum_i^N\hbar\sqrt{2\nu m_N^{-1}}\left(n+\frac{3}{2}\right)=3N\hbar\sqrt{\frac{\nu}{2m_N}}$$
In the *__spectrum__*, the first excited state's energy is 
$$\Delta E:=\hbar\sqrt{2\nu m_N^{-1}}$$
above the ground state.
For $m(\pi)=806\,$MeV$\,\to m_N=1632\,$MeV, and $\hbar c=197.3161329\,$MeVfm,
$$E_0\left(N=2,\nu=\frac{2m_N}{(\hbar c)^2}=0.0839376\,\text{MeVfm}\right)=6\,\text{MeV}$$and
$$\Delta E(.,.)=1\,\text{MeV}$$

In [8]:
status, output = commands.getstatusoutput('./svm.x np_3dosci_806')
outp = float([ line for line in open('./output/np_3dosci_806.txt')][-2].split('E=')[1])
print('E(0,S=1) = %4.4f MeV'%outp)

E(0,S=1) = -19.3656 MeV
E(0,S=0) = -15.7653 MeV


### $$\sum_i^Nq(i)\left(x^2(i)+y^2(i)+z^2(i)\right)$$

*ECCE*


In [9]:
#status, output = commands.getstatusoutput('./svm.x np_triplet_4_806')
#outp = float([ line for line in open('./output/np_triplet_4_806.txt')][-2].split('E=')[1])
#print('E(0,S=1) = %4.4f MeV'%outp)

### $$\sum_i^Nq(i)\left(x^2(i)+y^2(i)+\epsilon z^2(i)\right)$$

*ECCE*

In [10]:
#status, output = commands.getstatusoutput('./svm.x np_triplet_4_806')
#outp = float([ line for line in open('./output/np_triplet_4_806.txt')][-2].split('E=')[1])
#print('E(0,S=1) = %4.4f MeV'%outp)

### $$\sum_i^Nq(i)\sigma^z(i)$$

*ECCE*

In [11]:
#status, output = commands.getstatusoutput('./svm.x np_triplet_4_806')
#outp = float([ line for line in open('./output/np_triplet_4_806.txt')][-2].split('E=')[1])
#print('E(0,S=1) = %4.4f MeV'%outp)